optuna/visualization/_rank.py

```diff
@@ -160,16 +169,18 @@ def target(trial: FrozenTrial) -> float:
     target_values = np.array([target(trial) for trial in trials])
     raw_ranks = _get_order_with_same_order_averaging(target_values)
     color_idxs = raw_ranks / (len(trials) - 1) if len(trials) >= 2 else np.array([0.5])
```

Comment:
Currently, the rank does not take constraints into consideration, so when there are many many "good" points that do not satisfy constraints, points will be colored in either gray or "bad" colors.
A more intuitive alternative is to color points based on ranks of feasible points only. What do you think?

Reply:
@contramundum53 Sounds great, but there were two things that concerned me.

1. I think it's important that whether user can notice constraints are appropriately effective. I believe `plot_rank` is one of such ways. For example, below cases result same visualization if we make colors based on only feasible points.

- N infeasible points with target value smaller than $val$, M feasible points with target value larger than $val$
- N infeasible points with target value larger than $val$, M feasible points with target value smaller than $val$

2. How to treat `tick_info` is very difficult. It's so strange for me that there may be some infeasible points with out of `tick_info` range.

If aboves are no problems for you, i would change base colors(But is it really necessary to plot infeasible trials in such case?).

Reply:
Hmm, I see there could be controversy. Let's merge this PR and see how users feel 👍